dev_docs/contributing/documentation.mdx

```diff
@@ -0,0 +1,195 @@
+
+Do:
+
+```ts
+export interface AnInterface { bar: string };
+export type foo: string | AnInterface;
+```
+
+Don't:
+
+```ts
+interface AnInterface { bar: string };
+export type foo: string | AnInterface;
+```
+
+### Avoid “Pick”
+
+`Pick` not only ends up being unhelpful in our documentation system, but it's also of limited help in your IDE. For that reason, avoid `Pick` and other similarly complex types on your public API items. Using these semantics internally is fine.
+
+![pick api documentation](../assets/api_doc_pick.png)
+
+
+### Debugging tips
+
+There are three great ways to debug issues with the API infrastructure.
+
+1. Write a test
+
+[api_doc_suite.test.ts](https://github.com/elastic/kibana/blob/master/packages/kbn-docs-utils/src/api_docs/tests/api_doc_suite.test.ts) is a pretty comprehensive test suite that builds the test docs inside the [**fixtures** folder](https://github.com/elastic/kibana/tree/master/packages/kbn-docs-utils/src/api_docs/tests/__fixtures__/src).
+
+Edit the code inside `__fixtures__` to replicate the bug, write a test to track what should happen, then run `yarn jest api_doc_suite`.
+
+Once you've verified the bug is reproducible, use debug messages to narrow down the problem. This is much faster than running the entire suite to debug.
+
+2. Use [ts-ast-viewer.com](https://ts-ast-viewer.com/#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgMQgjgG8AoOSudJAfgC44AKdIxgZximQHMBKOAF4AfHE7ckPANxkAvkA)
+
+This nifty website will let you add some types and see how the system parses it. For example, the link above shows there is a `QuestionToken` as a sibling to the `FunctionType` which is why [this bug](https://github.com/elastic/kibana/issues/107145) reported children being lost. The API infra system didn't categorize the node as a function type node.
+
+3. Play around with `ts-morph` in a Code Sandbox.
+
+You can fork [this Code Sandbox example](https://codesandbox.io/s/typescript-compiler-issue-0lkwx?file=/src/use_ts_compiler.ts) that was used to explore how various ways to create the node signature returns different results (e.g. `node.getType.getText()` shows different results than `node.getType.getText(node)`).  Here is [another messy example](https://codesandbox.io/s/admiring-field-5btxs).
+
+The code sandbox approach can be a lot faster to iterate compared to running it in Kibana.
+
+## Example plugins
+
+Running Kibana with `yarn start --run-examples` will include all [example plugins](https://github.com/elastic/kibana/tree/main/examples). These are tested examples of platform services in use. We strongly encourage anyone providing a platform level service or <DocLink id="kibBuildingBlocks" text="building block"/> to include a tutorial that links to a tested example plugin. This is better than relying on copied code snippets, which can quickly get out of date.
+
+You can also visit these [examples plugins hosted online](https://demo.kibana.dev/8.2/app/home). Note that because anonymous access is enabled, some
```

Comment:
This links to a specific version (8.2) which I assume is outdated quickly (e.g. https://demo.kibana.dev/7.10/app/home does not work). Is there a possibility to keep this up-to-date automatically? (e.g. a variable here in the `.mdx` file or using a different URL structure in the nginx hosting https://demo.kibana.dev/)?

Reply:
Not yet, but filed https://github.com/elastic/kibana/issues/126772 so we can hopefully get this.